packages/labs/analyzer/src/lib/javascript/jsdoc.ts

```diff
@@ -109,34 +131,49 @@ export const parseNamedTypedJSDocInfo = (tag: ts.JSDocTag) => {
  * * @slot name description
  * * @slot name - description
  * * @slot name: description
+ * * @cssProp [--name=default]
+ * * @cssProp [--name=default] description
+ * * @cssProp [--name=default] - description
+ * * @cssProp [--name=default]: description
  */
 export const parseNamedJSDocInfo = (
   tag: ts.JSDocTag,
   requireDash = false
-): NamedDescribed | undefined => {
+): NamedDescribedDefault | undefined => {
   const comment = getJSDocTagComment(tag);
   if (comment == undefined) {
     return undefined;
   }
-  const nameDesc = comment.match(
-    requireDash ? parseNameDashDescRE : parseNameDescRE
-  );
-  if (nameDesc === null) {
-    throw new DiagnosticsError(
-      tag,
-      `Unexpected JSDoc format.${
-        parseNameDashDescRE
-          ? ` Tag must contain a whitespace-separated dash between the name and description, i.e. '@slot header - This is the description'`
-          : ''
-      }`
+  if (comment.match(/^\[.+=.+\]/)) {
```

Comment:
this should be a `test` rather than a `match` really, e.g. `/^whatever/.test(comment)` and then you can maybe put the regex in a const somewhere too

Reply:
im wondering if we should also just make this a special case for css props rather than matching syntactically

we could just check if the tag is `cssProp` and run this alternative logic (which i suppose would also line up with my other suggestion of supporting `cssProp --foo`, since this regex wouldn't allow that right now)

Reply:
>  supporting cssProp --foo, since this regex wouldn't

Help me understand this, since the test cases include this case.

https://github.com/lit/lit/blob/3e510743d314bc7399aec2a45cfd7a67e2cfa589/packages/labs/analyzer/test-files/ts/vanilla-jsdoc/src/element-a.ts#L27

Do you mean you'd prefer to not branch here? Me too. I initially tried to do it in one big honking regexp but that seemed to be prohibitively finicky, so I backed it off to an `if else`. I'm not sure how we'd support both `cssProp --foo` and `cssProp [--foo=bar]` without branching but also with a reasonably readable regexp (= superhero name up for grabs)

WDYT?

Reply:
i misunderstood partially, i didn't realise the `else` is what covers `@cssProp --foo`